### PR TITLE
Updates telecommunications interceptor for circuits.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Removes the ability for the circuit interceptor to jam communications.
T-Comm Interceptor now requires encryption keys to access secure channels.

## Why It's Good For The Game

Prevents the creation of circuit jammers using circuits.
Prevents the creation of a global encrpytion key using circuits.

## Changelog
:cl:
balance: Circuit comms receiver can no longer jam comms.
balance: Circuit comms receiver requires encryption keys.
/:cl:

### CODE REVIEW

**Comment by ItsTisme** [comment]
Finaly

**Comment by bloons3** [comment]
Why did either of those exist in the first place 🤔

**Comment by francinum** [comment]
As a substitute for NTSL.

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 1]
```diff
@@ -1368,28 +1367,47 @@
...
+	else
+		to_chat(user, "<span class='notice'>There are no encryption keys to remove from the mechanism.</span>")
+
+/obj/item/integrated_circuit/input/tcomm_interceptor/proc/recalculate_channels()
+	whitelisted_freqs.Cut()
+	whitelisted_freqs.Add(FREQ_COMMON)
+	var/list/weakreffd_ekeys = list()
+	for(var/o in encryption_keys)
+		var/obj/item/encryptionkey/K = o
+		weakreffd_ekeys += WEAKREF(K)
```
Comment (park66665) [inline code/modules/integrated_electronics/subtypes/input.dm]: What is the purpose of having a local list of weakref to keys? Do you plan to use it?

Comment (PowerfulBacon) [reply code/modules/integrated_electronics/subtypes/input.dm]: for ejecting the keys

Comment (PowerfulBacon) [reply code/modules/integrated_electronics/subtypes/input.dm]: oh wait nevermind I have no fucking idea

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 2]
```diff
@@ -1368,28 +1367,47 @@
...
+		for(var/i in K.channels)
+			whitelisted_freqs |= GLOB.radiochannels[i]
```
Comment (PowerfulBacon) [inline code/modules/integrated_electronics/subtypes/input.dm]:
```suggestion
/obj/item/integrated_circuit/input/tcomm_interceptor/proc/recalculate_channels()
	whitelisted_freqs.Cut()
	whitelisted_freqs.Add(FREQ_COMMON)
	for(var/o in encryption_keys)
		var/obj/item/encryptionkey/K = o
		for(var/i in K.channels)
			whitelisted_freqs |= GLOB.radiochannels[i]
```

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 3]
```diff
@@ -1352,12 +1352,11 @@
 
 /obj/item/integrated_circuit/input/tcomm_interceptor/Destroy()
 	qdel(receiver)
-	GLOB.ic_jammers -= src
 	..()
```
Comment (park66665) [inline code/modules/integrated_electronics/subtypes/input.dm]: At least `Cut()` the `encryption_keys` list. If you feel brave, `QDEL_LIST` it or just let it drop all the stored encryption keys.

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 4]
```diff
@@ -1368,28 +1367,45 @@
...
+		encryption_keys += O
+		recalculate_channels()
+		to_chat(user, "<span class='notice'>You slide \the [O] inside the circuit.</span>")
+	else
+		..()
+
+/obj/item/integrated_circuit/input/tcomm_interceptor/attack_self(mob/user)
+	if(encryption_keys.len)
+		for(var/i in encryption_keys)
+			var/obj/O = i
```
Comment (park66665) [inline code/modules/integrated_electronics/subtypes/input.dm]:
```suggestion
		for(var/obj/item/encryptionkey/O as() in encryption_keys)
```

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 5]
```diff
@@ -1368,28 +1367,45 @@
...
+		to_chat(user, "<span class='notice'>You slide the encryption keys out of the circuit.</span>")
+		recalculate_channels()
+	else
+		to_chat(user, "<span class='notice'>There are no encryption keys to remove from the mechanism.</span>")
+
+/obj/item/integrated_circuit/input/tcomm_interceptor/proc/recalculate_channels()
+	whitelisted_freqs.Cut()
+	whitelisted_freqs.Add(FREQ_COMMON)
+	for(var/o in encryption_keys)
+		var/obj/item/encryptionkey/K = o
```
Comment (park66665) [inline code/modules/integrated_electronics/subtypes/input.dm]:
```suggestion
	for(var/obj/item/encryptionkey/K as() in encryption_keys)
```

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 6]
```diff
@@ -1352,12 +1352,11 @@
 
 /obj/item/integrated_circuit/input/tcomm_interceptor/Destroy()
 	qdel(receiver)
```
Comment (park66665) [inline code/modules/integrated_electronics/subtypes/input.dm]:
```suggestion
	QDEL_NULL(receiver)
```

**Comment by Sarchutar** [comment]
Can you also remove everything with `GLOB.remote_control` since it has no use, atleast that i can see, without the jamming function?

**Pull request merged**